### PR TITLE
Drop custom json encoding for tool test API endpoint

### PR DESCRIPTION
**Pull request by nvk747**

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/tools.py [hunk 1]
```diff
@@ -205,7 +206,7 @@ def test_data(self, trans, id, **kwd):
         def json_encodeify(obj):
             if isinstance(obj, OrderedDict):
                 return dict(obj)
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/tools.py]: This doesn't seem to be needed after d6cf262ffe9c7c0a4eac4e47aa750340c8f0010e

File: lib/galaxy/webapps/galaxy/api/tools.py [hunk 2]
```diff
@@ -205,7 +206,7 @@ def test_data(self, trans, id, **kwd):
...
-            elif isinstance(obj, map):
+            elif inspect.isclass(map) and isinstance(obj, map):
                 return list(obj)
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/tools.py]: @mvdbeek This also shouldn't be needed any more after https://github.com/galaxyproject/galaxy/pull/9416 ?

I think we may just drop the method.

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/tools.py]: Yep, done that. Targeting dev for more extended test coverage, we can backport after merge

**Comment by mvdbeek** [comment]
@galaxybot test this

**Pull request merged**